Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+PathComponent.swift

```diff
@@ -13,7 +13,16 @@ import SymbolKit
 /// All known symbol kind identifiers.
 ///
 /// This is used to identify parsed path components as kind information.
-private let knownSymbolKinds = Set(SymbolGraph.Symbol.KindIdentifier.allCases.map { $0.identifier })
+private let knownSymbolKinds = Set<String>(
```

Comment:
I just found that there is a `SymbolGraph.Symbol.KindIdentifier.register()` function but these extended symbol kinds don't seem to be registered anywhere. 

Since I need to know them here I'll add the registration here.

Reply:
Sounds good. Maybe we should file a bug on symbolkit to be add them as official kinds? But this looks good for now.

Reply:
I vaguely remember there being some discussion around this and a conclusion that these kinds are specific to how DocC wants to present extensions and extended modules and that the Swift compiler uses another structure for what it emits in the symbol graph file. @theMomax would likely know more.

Reply:
@d-ronnqvist you're right, those types shouldn't be registered with SymbolKit. The registration is only required for (de)serialisation, however, the kinds we're talking about here are for internal use within swift-docc only, and should not be contained (and therefore also not recognised) in any symbol graph files given to swift-docc as input.